### ISSUE TITLE
Unable to train ANN network

### ISSUE THREAD

**Issue by umerehsan66** [nilmtk/nilmtk#785]
Hi 

I am using REDD data set. I have created dataframe of the following form
![query](https://user-images.githubusercontent.com/48155883/59066848-6a9d2b80-88c9-11e9-906a-44450a9e8541.jpg)
First Column is Main Power. 2nd column is Fridge power.
My ANN is created through following code
classifier = Sequential()
classifier.add(Dense(units=6, kernel_initializer='uniform', activation='relu', input_dim=1)) #input and 1st hidden layer 
classifier.add(Dense(units=6, kernel_initializer='uniform', activation='relu')) #2nd hidden layer 
classifier.add(Dense(units=600, kernel_initializer='uniform', activation='softmax')) #output layer
classifier.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy']) #compiling 
classifier.fit(X_train, Y_train, batch_size=10, epochs=100) #fitting training set

The idea is to train the ANN on main power as input and fridge power as output. The code does run but training accuracy is 3.4 %.
I am really stuck here and beginner in data science. please provide your valuable input to improve this
@nipunbatra @rakim2015

**Comment by umerehsan66** [comment]
Also before Initializing ANN, feature scaling above mentioned dataframe was applied through
X_train = sc.fit_transform(X_train.reshape(-1,1))

**Comment by rajatgupta1234** [comment]
@umerehsan66 
`classifier.add(Dense(units=6, kernel_initializer='uniform', activation='relu', input_dim=1)) #input and 1st hidden layer`

Here input_dim=size of input which can't be 1, you have to specify  the numbers of mains data rows for training.

**Comment by PMeira** [comment]
Since this is off-topic and is already handled, closing it.
